src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/ThrowHelper.cs

```diff
@@ -1,26 +1,97 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Microsoft.Quic;
+using System.Security.Authentication;
+using static Microsoft.Quic.MsQuic;
+
 namespace System.Net.Quic.Implementations.MsQuic
 {
     internal static class ThrowHelper
     {
-        internal static Exception GetConnectionAbortedException(long errorCode)
+        internal static QuicException GetConnectionAbortedException(long errorCode)
         {
             return errorCode switch
             {
-                -1 => new QuicOperationAbortedException(), // Shutdown initiated by us.
-                long err => new QuicConnectionAbortedException(err) // Shutdown initiated by peer.
+                -1 => GetOperationAbortedException(), // Shutdown initiated by us.
+                long err => new QuicException(QuicError.ConnectionAborted, SR.Format(SR.net_quic_connectionaborted, err), err, null) // Shutdown initiated by peer.
             };
         }
 
-        internal static Exception GetStreamAbortedException(long errorCode)
+        internal static QuicException GetStreamAbortedException(long errorCode)
         {
             return errorCode switch
             {
-                -1 => new QuicOperationAbortedException(), // Shutdown initiated by us.
-                long err => new QuicStreamAbortedException(err) // Shutdown initiated by peer.
+                -1 => GetOperationAbortedException(), // Shutdown initiated by us.
+                long err => new QuicException(QuicError.StreamAborted, SR.Format(SR.net_quic_streamaborted, err), err, null) // Shutdown initiated by peer.
             };
         }
+
+        internal static QuicException GetOperationAbortedException(string? message = null)
+        {
+            return new QuicException(QuicError.OperationAborted, message ?? SR.net_quic_operationaborted, null, null);
+        }
+
+        internal static Exception GetExceptionForMsQuicStatus(int status, string? message = null)
+        {
+            //
+            // Start by checking for statuses mapped to QuicError enum
+            //
+
+            if (status == QUIC_STATUS_ADDRESS_IN_USE) return new QuicException(QuicError.AddressInUse, SR.net_quic_address_in_use, null, null);
```

Comment:
Where does the msquic status get set to the `IOException.HResult`?

Reply:
it is set for the InternalError case at the end of the method

Reply:
I thought we're gonna set it in all cases, not just for `InternalError`? In some cases, we still combine multiple statuses in one exception so it might be useful for diagnostics.

Reply:
status is still platform specific, right? e.g. the actual value would differ between Windows and Linux.
Also would setting the `HResult` change text of the exception? If so, I would do it only for cases when useful and I would depend on `QuicError` in other cases.

Reply:
> status is still platform specific, right?

Yes.

> Also would setting the HResult change text of the exception?

I don't see how that would happen, the `HResult` setter is just setting a field.